kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -289,3 +407,224 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
+    ypc = pc[..., 1]
+    L = pc[..., 2]
+
+    # Detector coordinates in microns
+    det_x = (
+        -((-xpc - (1.0 - detector.ncols) * 0.5) - np.arange(0, detector.ncols))
+        * detector.px_size
+    )
+    det_y = (
+        (ypc - (1.0 - detector.nrows) * 0.5) - np.arange(0, detector.nrows)
+    ) * detector.px_size
+
+    # Auxilliary angle to rotate between reference frames
+    theta_c = np.radians(detector.tilt)
+    sigma = np.radians(detector.sample_tilt)
+
+    alpha = (np.pi / 2) - sigma + theta_c
+    ca = np.cos(alpha)
+    sa = np.sin(alpha)
+
+    # NYI
+    omega = np.radians(0)  # angle between normal of sample and detector
+    cw = np.cos(omega)
+    sw = np.sin(omega)
+
+    r_g_array = np.zeros((detector.nrows, detector.ncols, 3))
+
+    Ls = -sw * det_x + L * cw
+    Lc = cw * det_x + L * sw
+
+    i, j = np.meshgrid(
+        np.arange(detector.nrows - 1, -1, -1),
+        np.arange(detector.ncols),
+        indexing="ij",
+    )
+
+    r_g_array[..., 0] = det_y[i] * ca + sa * Ls[j]
+    r_g_array[..., 1] = Lc[j]
+    r_g_array[..., 2] = -sa * det_y[i] + ca * Ls[j]
+    r_g = Vector3d(r_g_array)
+
+    return r_g.unit
+
+
+def _get_lambert_interpolation_parameters(
+    rotated_direction_cosines: Vector3d,
+    scale: Union[int, float],
+    npx: int,
+    npy: int,
+) -> tuple:
+    """Get Lambert interpolation parameters as described in EMsoft.
+
+    Parameters
+    ----------
+    rotated_direction_cosines : Vector3d
+        Rotated direction cosines vector.
+    scale : int
+        Factor to scale up from Rosca-Lambert projection to the master pattern.
+    npx : int
+        Number of pixels on the master pattern in the x direction.
+    npy : int
+        Number of pixels on the master pattern in the y direction.
+
+    Returns
+    ----------
+    nii : numpy.ndarray
+        Row coordinate of a point.
+    nij : numpy.ndarray
+        Column coordinate of a point.
+    niip : numpy.ndarray
+        Row coordinate of neighboring point.
+    nijp : numpy.ndarray
+        Column coordinate of a neighboring point.
+    di : numpy.ndarray
+        Interpolation weight factor.
+    dj : numpy.ndarray
+        Interpolation weight factor.
+    dim : numpy.ndarray
+        Interpolation weight factor.
+    djm : numpy.ndarray
+        Interpolation weight factor.
+    """
+    # Normalized direction cosines to Rosca-Lambert projection
+    xy = (
+        scale
+        * LambertProjection.project(rotated_direction_cosines)
+        / (np.sqrt(np.pi / 2))
+    )
+
+    i = xy[..., 0]
+    j = xy[..., 1]
+    nii = i.astype(int) + scale
+    nij = j.astype(int) + scale
+    niip = nii + 1
+    nijp = nij + 1
+    niip = np.where(niip < npx, niip, nii)
+    nijp = np.where(nijp < npy, nijp, nij)
+    nii = np.where(nii < 0, niip, nii)
+    nij = np.where(nij < 0, nijp, nij)
+    di = i - nii + scale
+    dj = j - nij + scale
+    dim = 1.0 - di
+    djm = 1.0 - dj
+
+    return (
+        nii.astype(int),
```

Comment:
Would it work to make these int after `np.where(niip < npx, niip, nii).astype(int)`, and then get all returns on one line?